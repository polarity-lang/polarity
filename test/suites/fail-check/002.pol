data Bool { True: Bool, False: Bool }

data Nat {
    Z: Nat,
    S(n: Nat) : Nat,
}

data Foo(b: Bool) {
    Bar : Foo(True),
    Baz : Foo(False),
}

def Foo(True).foo() : Nat {
    Bar() => Z,
    Baz() absurd,
}

Baz.foo()
