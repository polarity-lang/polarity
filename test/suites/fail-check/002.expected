T-002

  × The following terms are not equal:
  │   1: Foo(False)
  │   2: Foo(True)
  │ 
    ╭─[002.pol:10:11]
  9 │     Bar : Foo(True),
 10 │     Baz : Foo(False),
    ·           ─────┬────
    ·                ╰── Source of (1)
 11 │ }
 12 │ 
 13 │ def Foo(True).foo() : Nat {
    ·     ────┬────
    ·         ╰── Source of (2)
 14 │     Bar() => Z,
 15 │     Baz() absurd,
 16 │ }
 17 │ 
 18 │ data Unit { Top }
 19 │ 
 20 │ def Unit.example : Nat {
 21 │     Top => Baz.foo
    ·            ─┬─
    ·             ╰── While elaborating
 22 │ }
    ╰────
